Lizarov, Anna. Assignment#2.Rmd

```diff
@@ -0,0 +1,163 @@
+names(V.TCH) <- c("id", "gender", "topic", "group")
+names(V.STD) <- c("id", "gender", "topic", "group")
+
+#Bind the two data frames together (you will get a warning because the teacher data frame has 5 types of id (A,B,C,D,E) and the student has 25 (1-30), this isn't a problem)
+VERTEX <- dplyr::bind_rows(V.TCH, V.STD)
+```
+```{r}
+#Load the igraph package
+
+library(igraph)
+
+#First we will make an object that contains the graph information using our two dataframes EDGE and VERTEX. Notice that we have made "directed = TRUE" - our graph is directed since discipline is being given from a teacher to a student.
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
+
+plot(g,layout=layout.fruchterman.reingold)
+
+#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+```
+# Part II
+```{r}
+library(tidyr)
+# Disciplinary action count for students
+total_st_EDGE <- EDGE %>% group_by(to) %>% summarize(sum(count)) 
+names(total_st_EDGE) <- c("id","count")
+# Disciplinary action count for teachers
+total_t_EDGE <- EDGE %>% group_by(from) %>% summarize(sum(count))
+names(total_t_EDGE) <- c("id","count")
+total <- bind_rows(total_st_EDGE, total_t_EDGE)
+VERTEX <- full_join(VERTEX, total, by="id")
+# Graph
+gr <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot(gr,layout=layout.fruchterman.reingold, 
+     vertex.color=as.factor(VERTEX$gender), 
+     edge.arrow.size = 0.5,
+     edge.width=EDGE$count,
+     vertex.size = VERTEX$count*2)
+```
+
+# Part III
+```{r}
+(cd <- read.csv("HUDK4050-2018-classes.csv", header = TRUE))
+```
+```{r}
+(cd2 <- select(cd, Q8, Q9, Q1, Q3, Q4, Q5, Q6, Q7))
+```
+```{r}
+colnames(cd2) <- as.character(unlist(cd2[1, ]))
+cd2 <- cd2[-(1:2), ]
+rownames(cd2) <- NULL
+cd2
+```
+```{r}
+names(cd2) <- gsub(" ", ".", names(cd2))
+cd2
+```
+```{r}
+cd3 <- cd2 %>% unite(First_Last, First.Name, Last.Name, sep="_", remove=TRUE) %>% print
+```
+```{r}
+cd4 <- cd3 %>% gather(Course_Label, Course_Name, Class.1, Class.2, Class.3,Class.4,Class.5, Class.6, na.rm = TRUE, convert = FALSE) %>% print
+```
+```{r}
+(cd4 <- select(cd4, First_Last, Course_Name))
+```
+
+```{r}
+# Everyone in this dataset is taking HUDK 4050, so I removed it. Otherwise, there is an error message when spreading the dataset.
+(cd4 <- filter(cd4, Course_Name>0, Course_Name != "HUDK 4050"))
+```
+
+```{r}
+cd4$Count <- 1
+```
+
+```{r}
+(cd5 <- spread(cd4, Course_Name, Count))
```

Comment:
At some point prior to this, you need to clean the data a little bit. Not everyone formatted the class names the same way. For example, your code does not treat "HUDK4050" and "HUDK 4050" as though they are the same class, although we know that they are.

You can use gsub (similar to how you used it above) to remove the spacing.